File: packages/models/src/models/feedback.py
from enum import StrEnum

from pydantic import BaseModel, Field


class FeedbackType(StrEnum):
    BUG = "bug"
    FEATURE = "feature"
    OTHER = "other"


class FeedbackRequest(BaseModel):
    feedback_type: FeedbackType
    message: str = Field(..., min_length=10, max_length=5000)


class FeedbackResponse(BaseModel):
    success: bool
    message: str

File: apps/web/src/server/functions/feedback.ts
import { createServerFn } from '@tanstack/react-start'
import { z } from 'zod'
import { authMiddleware } from '../middleware/auth'
import lazycloudApi from '../lazycloud-api'

const feedbackTypes = ['bug', 'feature', 'other'] as const

export const submitFeedback = createServerFn({ method: 'POST' })
  .middleware([authMiddleware])
  .inputValidator(
    z.object({
      feedbackType: z.enum(feedbackTypes),
      message: z
        .string()
        .min(10, 'Message must be at least 10 characters')
        .max(5000, 'Message is too long (max 5000 characters)'),
    }),
  )
  .handler(async ({ context, data }) => {
    return lazycloudApi.submitFeedback(
      context.accessToken,
      data.feedbackType,
      data.message,
      'web',
    )
  })
